ddtrace/contrib/internal/requests/connection.py

```diff
@@ -71,80 +68,83 @@ def _extract_query_string(uri):
+
+    # Determine service name (check pin config for "service" or "service_name")
     service = None
-    if cfg["split_by_domain"] and hostname:
-        service = hostname
-    if service is None:
-        service = cfg.get("service", None)
+    if integration_config and isinstance(integration_config, dict):
+        service = integration_config.get("service") or integration_config.get("service_name")
     if service is None:
-        service = cfg.get("service_name", None)
-    if service is None:
-        service = trace_utils.ext_service(None, config.requests)
-
-    operation_name = schematize_url_operation("requests.request", protocol="http", direction=SpanDirection.OUTBOUND)
-    with tracer.trace(operation_name, service=service, resource=f"{method} {path}", span_type=SpanTypes.HTTP) as span:
-        span._set_tag_str(COMPONENT, config.requests.integration_name)
-
-        # set span.kind to the type of operation being performed
-        span._set_tag_str(SPAN_KIND, SpanKind.CLIENT)
-
-        # PERF: avoid setting via Span.set_tag
-        span.set_metric(_SPAN_MEASURED_KEY, 1)
-
-        # propagate distributed tracing headers
-        if cfg.get("distributed_tracing"):
-            HTTPPropagator.inject(span.context, request.headers)
-
-        response = response_headers = None
+        service = trace_utils.ext_service(pin, config.requests)
+
+    # AppSec Integration: Provide context items for RASP and API Security (API10)
+    # The requests library calls urllib3 methods that have AppSec wrappers (see _common_module_patches.py).
+    # These wrappers expect to find 'full_url' and 'use_body' via core.get_item() to:
+    #   - full_url: Analyze outbound requests for SSRF threats (e.g., requests to 169.254.169.254)
+    #   - use_body: Determine if response bodies should be analyzed for sensitive data leakage
+    # Without these context items, AppSec security checks are skipped for requests made via this library.
+    full_url = url
+    use_body = False
+    try:
+        from ddtrace.appsec._asm_request_context import _get_asm_context
+        from ddtrace.appsec._asm_request_context import should_analyze_body_response
+
+        if asm_context := _get_asm_context():
+            use_body = should_analyze_body_response(asm_context)
+    except ImportError:
+        # AppSec is not available, which is fine - not all users have it enabled
+        pass
+
+        # the above should be handled by a asm event - dispatch the event and use the result somehow
+
+    with core.context_with_data(
+        "requests.send",
+        span_name=schematize_url_operation("requests.request", protocol="http", direction=SpanDirection.OUTBOUND),
+        pin=pin,
+        service=service,
+        resource=f"{method} {path}",
+        span_type=SpanTypes.HTTP,
+        integration_config=integration_config,
+        measured=True,
+        tags={COMPONENT: config.requests.integration_name, SPAN_KIND: SpanKind.CLIENT},
+        request=request,
```

Comment:
We should avoid passing objects to the core with data that are not common across integrations. request here is specific to this library and can not be reused in like the httpx integration